src/topology/metric_space/basic.lean

```diff
@@ -863,6 +863,12 @@ lemma closed_ball_Icc {x r : ℝ} : closed_ball x r = Icc (x-r) (x+r) :=
 by ext y; rw [mem_closed_ball, dist_comm, real.dist_eq,
   abs_sub_le_iff, mem_Icc, ← sub_le_iff_le_add', sub_le]
 
+-- This lemma could go somewhere else, but I did not find a better fit for it.  Feel
+-- free to suggest where it may be added.  Here, all the needed imports are available!
+lemma mem_Icc_iff_abs_le {R : Type*} [linear_ordered_add_comm_group R] {x y z : R} :
+  abs (x - y) ≤ z ↔ y ∈ Icc (x - z) (x + z) :=
+abs_le.trans $ (and_comm _ _).trans $ and_congr sub_le neg_le_sub_iff_le_add
```

Comment:
Note that `sub_mem_Icc_iff_right` is now almost the right lemma here, but you then need to rewrite  `x - -z` into `x + z`.

Of course, you could just drop this lemma and use the pieces directly now.

Reply:
Below is my very clumsy attempt at using lemma `sub_mem_Icc_iff_right`: it is not simpler than what you initially found.

I really think that symmetric intervals are at least as important as random intervals, since they are the ones that arise from balls centred at the point.  For this reason, I would prefer to maintain the lemma above.

begin
  rw [← neg_neg z] { occs := occurrences.pos [3] },
  rw [← sub_eq_add_neg x (- z)],
  exact (abs_le.trans mem_Icc.symm).trans sub_mem_Icc_iff_right,
end